src/main/kotlin/pro/streem/sdk/Streem.kt

```diff
@@ -1,14 +1,31 @@
 package pro.streem.sdk
 
-public class Streem private constructor(
-    private val config: Config
-) {
+public class Streem private constructor(private val config: Config) {
+
+    /**
+     * Returns a [TokenBuilder] that can be used to create a Streem Token.
+     *
+     * This method is optimized for use from Java code. For Kotlin code, prefer the [buildToken] method.
+     *
+     * @param userId The user ID of the user to authenticate with this token.
+     */
     public fun tokenBuilder(userId: String): TokenBuilder = TokenBuilder(config, userId)
 
-    public fun buildToken(userId: String, block: TokenBuilder.() -> Unit): String =
-        tokenBuilder(userId).also(block).build()
+    /**
+     * Returns a string representation of a Streem Token configured with [builder].
+     *
+     * This method is optimized for use from Kotlin code. For Java code, prefer the [tokenBuilder] method.
+     *
+     * @param userId The user ID of the user to authenticate with this token.
+     */
+    public fun buildToken(userId: String, builder: StreemToken.() -> Unit): String =
```

Comment:
Yikes, `tokenBuilder` and `buildToken`?  I understand why you did it, but the public API is a bit confusing to look at.  Would it make more sense to use overloading, and have the same function name?  Or no?

Reply:
Yeah I'd debated that as well. We could name both methods `buildToken()` with overloads, like this:

```kt
    public fun buildToken(userId: String): TokenBuilder
    public fun buildToken(userId: String, builder: StreemToken.() -> Unit): String
```

But then the Java API feels slightly odd to me (not terribly so, but not ideal) because you're calling `buildToken()` and then calling `build()` on the result:

```java
var token = streem.buildToken(user.id)
    .name(user.name)
    .email(user.email)
    ...
    .build()
```

What do you think? Does that feel better than having the two differently-named `buildToken()` and `tokenBuilder()` methods? I am leaning towards this approach now that I've written it out.

Reply:
I just pushed a version with this rename. I promised this to BBY today, so I'll go ahead and merge in a couple hours if I don't hear back by then.